File: framework/languages/java/e2e/SpotService/Shared/src/main/java/systems/zlink/e2e/spotservice/shared/Contracts.java
package systems.zlink.e2e.spotservice.shared;

import java.util.List;

public final class Contracts {
    public static final String ROUTE_CHANNEL = "spot.service.route";
    public static final String ROUTE_PACKET = "RouteReq";
    public static final String EGRESS_CHANNEL = "spot.service.egress";
    public static final String INGRESS_CHANNEL = "spot.service.ingress";
    public static final String SPOT_MESH = "spot.service.mesh";
    public static final String SPOT_NODE = "play";

    private Contracts() {
    }

    public record StateReq(String op) {
    }

    public record StateRes(
        String spotRid,
        String nodeRid,
        String value) {
    }

    public record StateMsg(String value) {
    }

    public record SlowReq(String value) {
    }

    public record OutboundReq(String value) {
    }

    public record OutboundRes(
        String spotRid,
        String nodeRid,
        String channelReply) {
    }

    public record OutboundMsg(String value) {
    }

    public record MeshMsg(String value) {
    }

    public record TimerActivityReq(String value) {
    }

    public record TimerActivityRes(
        String spotRid,
        String value) {
    }

    public record RouteReq(String value) {
    }

    public record RouteRes(
        String value,
        String nodeRid,
        String routeRid) {
    }

    public record ActorProfile(
        String displayName,
        int level,
        List<String> tags) {
    }

    public record ActorAuthReq(
        String actorId,
        ActorProfile profile) {
    }

    public record ActorAuthRes(
        String actorId,
        String nodeRid,
        int boundCount,
        String displayName,
        int level,
        List<String> tags) {
    }

    public record ActorJoinReq(
        String spotRid,
        ActorProfile profile,
        List<String> tags) {
    }

    public record ActorJoinRes(
        String actorId,
        String spotRid,
        String nodeRid,
        String displayName,
        int level,
        List<String> tags) {
    }

    public record LeaveActorReq(String actorId) {
    }

    public record LeaveActorRes(
        String actorId,
        boolean accepted) {
    }

    public record ActorEchoReq(
        String value,
        int seq,
        ActorProfile profile) {
    }

    public record ActorEchoRes(
        String actorId,
        String spotRid,
        String nodeRid,
        String value,
        int requestSeq,
        int handlerSeq,
        String displayName,
        int level,
        List<String> tags) {
    }

    public record ActorPushNotify(
        String actorId,
        String spotRid,
        String value,
        int requestSeq,
        int handlerSeq) {
    }

    public record EvidenceEntry(
        String marker,
        String nodeRid,
        String spotRid,
        String value) {
    }

    public record EvidenceWaitReq(
        List<String> containsAll,
        int timeoutMilliseconds) {
    }

    public record EvidenceSnapshot(
        String nodeRid,
        List<EvidenceEntry> entries) {
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Shared/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/Contracts.kt
package systems.zlink.e2e.kotlin.spotservice

class Contracts private constructor() {
    companion object {
        const val ROUTE_CHANNEL: String = "spot.service.route"
        const val ROUTE_PACKET: String = "RoutePing"
        const val EGRESS_CHANNEL: String = "spot.service.egress"
        const val INGRESS_CHANNEL: String = "spot.service.ingress"
        const val SPOT_MESH: String = "spot.service.mesh"
        const val SPOT_NODE: String = "play"
        const val MULTI_SPOT_NODE_A: String = "multi-node-a"
        const val MULTI_SPOT_NODE_B: String = "multi-node-b"
        const val MULTI_ROUTE_CHANNEL_A: String = "multi-route-a"
        const val MULTI_ROUTE_CHANNEL_B: String = "multi-route-b"
    }

    @JvmRecord
    data class StateRequest(val op: String)

    @JvmRecord
    data class StateReply(
        val spotRid: String,
        val nodeRid: String,
        val value: String
    )

    @JvmRecord
    data class StateCommand(val value: String)

    @JvmRecord
    data class SlowRequest(val value: String)

    @JvmRecord
    data class OutboundRequest(val value: String)

    @JvmRecord
    data class OutboundReply(
        val spotRid: String,
        val nodeRid: String,
        val channelReply: String
    )

    @JvmRecord
    data class OutboundCommand(val value: String)

    @JvmRecord
    data class MeshEvent(val value: String)

    @JvmRecord
    data class TimerActivity(val value: String)

    @JvmRecord
    data class TimerStatus(
        val spotRid: String,
        val value: String
    )

    @JvmRecord
    data class RoutePing(val value: String)

    @JvmRecord
    data class RoutePong(
        val value: String,
        val nodeRid: String,
        val routeRid: String
    )

    @JvmRecord
    data class MultiNodeCreateSpotRequest(
        val spotRid: String,
        val delta: Int
    )

    @JvmRecord
    data class MultiNodeCreateSpotReply(
        val spotRid: String,
        val nodeRid: String,
        val state: String,
        val value: Int
    )

    @JvmRecord
    data class MultiNodeStateRouteRequest(
        val spotRid: String,
        val delta: Int
    )

    @JvmRecord
    data class MultiNodeStateRequest(
        val operation: String,
        val delta: Int
    )

    @JvmRecord
    data class MultiNodeStateReply(
        val spotRid: String,
        val nodeRid: String,
        val value: Int
    )

    @JvmRecord
    data class EvidenceWaitRequest(
        val containsAll: List<String>,
        val timeoutMilliseconds: Int = 10_000
    )

    @JvmRecord
    data class ActorProfile(
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorAuthRequest(
        val actorId: String,
        val profile: ActorProfile
    )

    @JvmRecord
    data class ActorAuthReply(
        val actorId: String,
        val nodeRid: String,
        val boundCount: Int,
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class MultiBindRequest(
        val firstActorId: String,
        val secondActorId: String,
        val profile: ActorProfile
    )

    @JvmRecord
    data class MultiBindReply(
        val boundCount: Int
    )

    @JvmRecord
    data class DestroyActorRequest(
        val actorId: String
    )

    @JvmRecord
    data class DestroyActorReply(
        val actorId: String,
        val destroyed: Boolean
    )

    @JvmRecord
    data class LeaveActorRequest(
        val actorId: String
    )

    @JvmRecord
    data class LeaveActorReply(
        val actorId: String,
        val accepted: Boolean
    )

    @JvmRecord
    data class ActorJoinRequest(
        val spotRid: String,
        val profile: ActorProfile,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorJoinReply(
        val actorId: String,
        val spotRid: String,
        val nodeRid: String,
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorEchoRequest(
        val value: String,
        val seq: Int,
        val profile: ActorProfile
    )

    @JvmRecord
    data class SlowSessionRequest(
        val value: String,
        val delayMilliseconds: Int
    )

    @JvmRecord
    data class SlowSessionReply(
        val value: String
    )

    @JvmRecord
    data class ActorEchoReply(
        val actorId: String,
        val spotRid: String,
        val nodeRid: String,
        val value: String,
        val requestSeq: Int,
        val handlerSeq: Int,
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorPush(
        val actorId: String,
        val spotRid: String,
        val value: String,
        val requestSeq: Int,
        val handlerSeq: Int
    )

    @JvmRecord
    data class EvidenceEntry(
        val marker: String,
        val nodeRid: String,
        val spotRid: String,
        val value: String?
    )

    @JvmRecord
    data class EvidenceSnapshot(
        val nodeRid: String,
        val entries: List<EvidenceEntry>
    )
}
